### ISSUE TITLE
Intermittent exceptions related to torch during actor initialisation

### ISSUE THREAD

**Issue by johann-petrak** [ray-project/ray#5776]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04
- **Ray installed from (source or binary)**: pip install under per-user Anaconda installation
- **Ray version**: 0.7.4
- **Python version**: 3.6.8
- **Exact command to reproduce**: Intermittent

### Describe the problem

I am getting intermittent exceptions which seem to happen as the actors get initialised. The exceptions are mostly those shown below. I can run exactly the same code and everything works fine, then try to run it again and get those exceptions with several attempts in a row. 

I am running this code from within a terminal in the Ubuntu window manager. One observation seems to be that once I get those exceptions, running the code again from the same terminal will also get the exceptions again. However when I close the terminal and try from another terminal window, it may work fine again (but not always does). 

I am not using Pytorch myself in this code, but I am using Spacy which seems to at some point at least try to load torch.

```
2019-09-25 17:01:07,326	ERROR worker.py:1791 -- Failed to unpickle actor class 'RayActorRunner' for actor ID 09b7cb0801000000. Traceback:
Traceback (most recent call last):
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/function_manager.py", line 712, in _load_actor_class_from_gcs
    actor_class = pickle.loads(pickled_class)
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/spacy/__init__.py", line 12, in <module>
    from .cli.info import info as cli_info
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/spacy/cli/__init__.py", line 6, in <module>
    from .train import train  # noqa: F401
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/spacy/cli/train.py", line 16, in <module>
    from .._ml import create_default_optimizer
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/spacy/_ml.py", line 31, in <module>
    import torch.nn
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/torch/nn/__init__.py", line 1, in <module>
    from .modules import *  # noqa: F401
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/torch/nn/modules/__init__.py", line 1, in <module>
    from .module import Module
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/torch/nn/modules/module.py", line 6, in <module>
    from ..backends.thnn import backend as thnn_backend
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/torch/__init__.py", line 83, in <module>
    __all__ += [name for name in dir(_C)
NameError: name '_C' is not defined

2019-09-25 17:01:07,326	ERROR worker.py:1791 -- Traceback (most recent call last):
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/import_thread.py", line 140, in fetch_and_execute_function_to_run
    function({"worker": self.worker})
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/worker.py", line 2280, in register_class_for_serialization
    "worker"].get_serialization_context(job_id)
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/worker.py", line 255, in get_serialization_context
    _initialize_serialization(job_id)
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/worker.py", line 1238, in _initialize_serialization
    pyarrow.register_torch_serialization_handlers(serialization_context)
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/pyarrow_files/pyarrow/serialization.py", line 230, in register_torch_serialization_handlers
    import torch
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/torch/__init__.py", line 280, in <module>
    from .functional import *
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/torch/functional.py", line 2, in <module>
    import torch.nn.functional as F
AttributeError: module 'torch' has no attribute 'nn'

2019-09-25 17:01:07,327	ERROR worker.py:1791 -- Traceback (most recent call last):
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/worker.py", line 938, in _process_task
    self.reraise_actor_init_error()
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/worker.py", line 237, in reraise_actor_init_error
    raise self.actor_init_error
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/worker.py", line 942, in _process_task
    function_name, args)
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/worker.py", line 855, in _get_arguments_for_execution
    values = self.get_object(object_ids)
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/worker.py", line 553, in get_object
    final_results = self.retrieve_and_deserialize(plain_object_ids, 0)
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/worker.py", line 443, in retrieve_and_deserialize
    self.current_job_id)
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/worker.py", line 255, in get_serialization_context
    _initialize_serialization(job_id)
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/worker.py", line 1238, in _initialize_serialization
    pyarrow.register_torch_serialization_handlers(serialization_context)
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/pyarrow_files/pyarrow/serialization.py", line 230, in register_torch_serialization_handlers
    import torch
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/torch/__init__.py", line 83, in <module>
    __all__ += [name for name in dir(_C)
NameError: name '_C' is not defined
```

and 

```
2019-09-25 17:01:12,336	ERROR worker.py:1731 -- Possible unhandled error from worker: ray_TemporaryActor:__init__() (pid=18269, host=pflaume)
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/ray/pyarrow_files/pyarrow/serialization.py", line 230, in register_torch_serialization_handlers
    import torch
  File "/home/johann/software/anaconda3/lib/python3.6/site-packages/torch/__init__.py", line 83, in <module>
    __all__ += [name for name in dir(_C)
NameError: name '_C' is not defined
```

**Comment by pcmoritz** [comment]
Thanks for reporting! One possible explanation is that there are two different Python versions installed and they somehow clash. If that's not the case and you can reproduce it in a clean install, sending us a minimal version of the reproduction code and how you set up the environment (e.g. on EC2 if possible) would help us to look into the problem.

**Comment by johann-petrak** [comment]
If this would be because of two different Python versions, why is it only happening rarely though? 
If e.g. running the process for a worker would invoke a different python version that causes this error, why would it not happen every time? 
Sadly I was not able to make this reproducible. As I said it only happens rarely and it may be that it happens in situations where a bug in the code causes the worker(s) to throw an exception. 
I am not completely sure but when I looked into this at one point it looked as if there were Ray processes still lingering after such an error, could it be related to such a situation?
I will report back if it happens again and I found out more ...
